packages/monaco/src/browser/monaco-text-model-service.ts

```diff
@@ -58,7 +73,16 @@ export class MonacoTextModelService implements monaco.editor.ITextModelService {
 
     protected async loadModel(uri: URI): Promise<MonacoEditorModel> {
         await this.editorPreferences.ready;
-        const resource = await this.resourceProvider(uri);
+
+        let resource;
+        // Check whether additional resource provider is registered for a scheme.
+        if (this.textContentResourceProviders.has(uri.scheme)) {
+            // Ask resource provider for a resource
+            resource = await this.getProvidedTextResource(uri);
+        } else {
+            resource = await this.resourceProvider(uri);
```

Comment:
How about implementing a special `ResourceResolver` instead without any changes to `MonacoTextModelService`?

Reply:
I was thinking about it but I don'n know why, have not implemented it. Thanks for notifying.
I will try to rework it and remove changes in `MonacoTextModelService`.

Reply:
Will I remove `registerTextModelContentProvider` from `ITextModelService` interface or should leave as it is?

Reply:
let's leave it, we need to check whether it is used by Monaco internally before removing - it's out of scope of this PR

Reply:
Reworked